services/src/pro/datasets/postgres.rs

```diff
@@ -48,6 +58,7 @@ where
     <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
 {
     async fn list_datasets(&self, options: DatasetListOptions) -> Result<Vec<DatasetListing>> {
+        self.update_datasets_status().await?;
```

Comment:
maybe add a comment here that we lazily delete uploads on the fly...
and a todo for a regular deletion task to replace this?

Reply:
I'm not sure I understand.

The requirement to update the datasets status is that the expiration date might have been reached. So before any operation, this needs to be checked. In that case, listing does have to do with dataset status.

I wrapped the functions into a lazy_dataset_updates and made a comment above it. Hopefully, this is what you wanted. 2865e61058c4a931cf41c73e497f2e812bc78b17